spec/models/household_spec.rb

```diff
@@ -4,5 +4,95 @@
   describe "relationships" do
     it { should have_many :chores }
     it { should have_many :members }
+    it { should have_many(:member_chores).through(:members) }
+  end
+
+  context "instance methods" do
+    describe "#randomize_chores" do
+      let!(:household) { create(:household) }
+      let!(:household2) { create(:household) }
+
+      let!(:member1) { create(:member, household:) }
+      let!(:member2) { create(:member, household:) }
+      let!(:member3) { create(:member, household:) }
+      let!(:member4) { create(:member, household: household2) }
+
+      let!(:chore1) { create(:chore, name: "chore1", household:, duration: 15) }
+      let!(:chore2) { create(:chore, name: "chore2", household:, duration: 15) }
+      let!(:chore3) { create(:chore, name: "chore3", household:, duration: 15) }
+      let!(:chore4) { create(:chore, name: "chore4", household:, duration: 30) }
+      let!(:chore5) { create(:chore, name: "chore5", household:, duration: 30) }
+      let!(:chore6) { create(:chore, name: "chore6", household:, duration: 30) }
+      let!(:chore7) { create(:chore, name: "chore7", household:, duration: 30) }
+      let!(:chore8) { create(:chore, name: "chore8", household:, duration: 45) }
+      let!(:chore9) { create(:chore, name: "chore9", household:, duration: 45) }
+      let!(:chore10) { create(:chore, name: "chore10", household: household2, duration: 45) }
+      let!(:member_chore10) { create(:member_chore, member: member4, chore: chore10) }
+
+      it "creates member chores that are evenly distributed" do
+        household.randomize_chores
+
+        Household.find(household.id).members.each do |member|
+          expect(total_chore_duration(member)).to be >= 60
+          expect(total_chore_duration(member)).to be <= 115
```

Comment:
I've never seen testing done this way for multiple objects, it seems way more efficient than writing the testing for each individually